### PR TITLE
Add boulder grade filter popover #238

### LINKED ISSUE #238: Add Bouldering grade filter popover
### Crag Finder story #178 
Use case:  as a boulderer I want to find bouldering areas with problems within my grade.

#### Tips:
The V grade component should be very similar to YDS filter and Radius filter. 
- YDS: https://github.com/OpenBeta/open-tacos/blob/develop/src/components/YDSFilter.tsx
- Radius: https://github.com/OpenBeta/open-tacos/blob/develop/src/components/RadiusFilter.tsx

Screen Shot 2022-03-24 at 8 37 31 PM

### PR DESCRIPTION
**Pull request by clintonlunn**
- [x] add bouldering filter range ui components
- [x] add BOULDER_DEFS with correct properties/values
- [x] move filter components into filter folder
- [x] add cragFilterStore boulderRange state value and mutation methods 
- [x] working map filtering
- [x] add mobile view props

### CODE REVIEW

**Comment by clintonlunn** [comment]
@vnugent I think I'm somewhat close here. Could you take a look and see if anything jumps out at you? I cannot get the same number of crags as I was getting before implementing this bouldering grade filter popover. It seems that it was displaying all bouldering areas. Now that there's a grade restriction on it, there are less crags (-8 less in the area I've been testing in, which is California). Maybe this is some bad data being omitted now?

Also, do we want to have separate methods for updating free vs bouldering climbing? i.e. like the methods i added in [index.ts](https://github.com/OpenBeta/open-tacos/pull/259/files#diff-d6e49eb2aa96fcb8102b97710b982cb62152810794869ba2d935d43469facebd)

**Comment by vnugent** [comment]
if you haven't done it already, install [Apollo Client Devtool](https://www.apollographql.com/docs/react/development-testing/developer-tooling/#apollo-client-devtools) browser extension, which allows you to browse cache data coming from the backend.  
FWIW, the graphql api returns all data by radius. No sport/trad/bouldering filtering is done there.

File: src/js/grades/rangeDefs.ts [hunk 1]
```diff
@@ -70,3 +70,67 @@ export const YDS_DEFS: SliderMarksType =
...
+  },
+  12: {
+    label: 'V11',
+    score: getScoreForSort('V11', GradeScales.VScale)
+  },
+  13: {
+    label: 'V12',
+    score: getScoreForSort('v12', GradeScales.VScale)
+  },
+  14: {
```
Comment (clintonlunn) [inline src/js/grades/rangeDefs.ts:132]: This is a part I am unsure about. I don't think will capture everything above v13+, but rather up to V13+. I am not sure how to account for this? Perhaps increase number of tick marks to account for v grades up to what is currently the max. A quick google suggests the highest grade out there is a v17 (https://www.99boulders.com/bouldering-grades)

**Comment by clintonlunn** [comment]
There is still a question about whether the filtering is happening correctly. I understand that the graphql requests happen only to select by radius and the other filtering methods only act on what has been retrieved in the radius based search.

**Comment by vnugent** [comment]
We can do more testing and address the data issue in a future PR. I'll have a look at the PR later.  Thanks!

**Pull request merged**